wandb/util.py

```diff
@@ -1552,6 +1553,63 @@ def load_as_json_file_or_load_dict_as_json(config: str) -> Any:
             return None
 
 
+def _parse_entity_project_item(path: str) -> tuple:
+    """Parses paths with the following formats: {item}, {project}/{item}, & {entity}/{project}/{item}.
+
+    Args:
+        path: `str`, input path; must be between 0 and 3 in length.
+
+    Returns:
+        tuple of length 3 - (item, project, entity)
+
+    Example:
+        alias, project, entity = _parse_entity_project_item("myproj/mymodel:best")
+
+        assert entity   == ""
+        assert project  == "myproj"
+        assert alias    == "mymodel:best"
+
+    """
+    words = path.split("/")
+    if len(words) == 0 or len(words) > 3:
+        raise ValueError(
+            "Invalid path: must be of the form {item}, {project}/{item}, or {entity}/{project}/{item}"
+        )
+    keys = ["item", "project", "entity"]
+    rev = reversed(words)
+    parsed = defaultdict(str, zip(keys, rev))
+    return tuple(parsed[key] for key in keys)
+
+
+def _resolve_aliases(aliases: Optional[Union[str, List[str]]]) -> List[str]:
+    """Takes in `aliases` which can be None, str, or List[str] and returns List[str].
+    Ensures that "latest" is always present in the returned list.
+
+    Args:
+        aliases: `Optional[Union[str, List[str]]]`
+
+    Returns:
+        List[str], with "latest" always present.
+
+    Example:
+        aliases = _resolve_aliases(["best", "dev"])
+        assert aliases == ["best", "dev", "latest"]
+
+        aliases = _resolve_aliases("boom")
+        assert aliases == ["boom", "latest"]
+
+    """
+    if aliases is None:
+        aliases = []
+    elif isinstance(aliases, str):
```

Comment:
do you need to check if aliases is of invalid type?

Reply:
nice -- I'll add this check.